lib/src/components/NxTable/NxTable.tsx

```diff
@@ -4,30 +4,34 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import React from 'react';
+import React, { useEffect, useRef, useState, useCallback } from 'react';
+import useMutationObserver from '@rooks/use-mutation-observer';
 import classnames from 'classnames';
 
-import {only} from '../../util/childUtil';
-import NxTableHead from './NxTableHead';
-import NxTableBody from './NxTableBody';
-import NxTableRow from './NxTableRow';
 import { ColumnCountContext } from './contexts';
 
 import { NxTableProps, nxTablePropTypes } from './types';
 export { NxTableProps };
 
+const mutationObserverConfig = { subtree: true, childList: true, attributes: false, characterData: false };
+
 const NxTable = function NxTableElement(props: NxTableProps) {
   const {className, children, ...attrs} = props,
-      thead = only(children, NxTableHead),
-      tbody = only(children, NxTableBody),
-      trow = thead && only(thead.props.children, NxTableRow),
-      columns = trow ? React.Children.count(trow.props.children) : 0;
+      tableRef = useRef<HTMLTableElement>(null),
+      [columnCount, setColumnCount] = useState(1),
+      updateColumnCount = useCallback(function updateColumnCount() {
+        if (tableRef.current) {
+          setColumnCount(tableRef.current.querySelectorAll('thead > tr:first-child > th').length);
+        }
+      }, []);
+
+  useEffect(updateColumnCount, []);
+  useMutationObserver(tableRef, updateColumnCount, mutationObserverConfig);
```

Comment:
This sets up a [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) that will trigger the column count update logic any time a DOM node is added or removed within the table.

I had initially gone with a simpler approach: `useEffect` with the `children` as its dependency, so that the logic re-runs every time the `children` changes.  I realized however that this was problematic in two ways:
1. it would re-run basically any time anything happened at the `NxTable` level, as the child nodes, even if semantically identical, would typically be new instances
2. it would fail to catch changes that result strictly from stateful logic within a child component.  The `MutationObserver` guarantees that not matter how the child DOM gets changed, we'll notice